internal/pkg/prune/prune.go

```diff
@@ -0,0 +1,248 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prune
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+	"k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/cli-experimental/internal/pkg/clik8s"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Prune prunes directories
+type Prune struct {
+	Client    client.Client
+	Out       io.Writer
+	Resources clik8s.ResourcePruneConfigs
+	Commit    *object.Commit
+}
+
+// Result contains the Prune Result
+type Result struct {
+	Resources clik8s.ResourcePruneConfigs
+}
+
+// Do executes the prune
+func (o *Prune) Do() (Result, error) {
+	if len(o.Resources) != 1 {
+		return Result{}, fmt.Errorf("prune only accepts one object, but got %v", o.Resources)
+	}
+	fmt.Fprintf(o.Out, "Doing `cli-experimental prune`\n")
+
+	u := o.Resources[0]
+	obj := &unstructured.Unstructured{}
+	obj.SetGroupVersionKind(u.GroupVersionKind())
+	err := o.Client.Get(context.Background(), client.ObjectKey{
+		Namespace: u.GetNamespace(),
+		Name:      u.GetName(),
+	}, obj)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			fmt.Fprintf(os.Stderr, "retrieving current configuration of %s from server for %v", u.GetName(), err)
+			return Result{o.Resources}, err
+		}
+		fmt.Fprintf(os.Stderr, "couldn't find a configmap for pruning %s", u.GetName())
```

Comment:
print the error here.  not necessarily true that the object couldn't be found - could be an auth error.

Reply:
updated